crds_data.ipynb

```diff
@@ -533,6 +533,95 @@
     "df.head()\n"
    ]
   },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Apply logbook flags"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 10,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def read_logbook(path):\n",
+    "    \"\"\"\n",
+    "    Read events in the logbook file (.csv) \n",
+    "    Drop empty lines.\n",
+    "    \n",
+    "    Dataframe columns name: Event | Start_Event | End_Event | Flags\n",
+    "    \n",
+    "    Parameters: path of .csv logbook\n",
+    "    Returns: dataframe with the logbook data\n",
+    "    \"\"\"\n",
+    "    lines = open(path_logbook).readlines()\n",
+    "    lines_skip = lines.index('Event,\"Start Event\\n')\n",
+    "    df = pd.read_csv(path, sep=',',  skiprows=lines_skip)\n",
+    "    df = df.rename(columns={\"Start Event\\ndd/mm/aaaa hh:mm\": \"Start_Event\",\n",
+    "                           \"End Event\\ndd/mm/aaaa hh:mm\": \"End_Event\"})\n",
+    "    df = df.dropna(how='all')\n",
+    "    return df"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 11,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "def drop_invalid_logbook(df):\n",
+    "    \"\"\"\n",
+    "    Drop invalid datatime in logbook dataframe.\n",
+    "    Index are reseted.\n",
+    "    \n",
+    "    Parameters: logbook dataframe\n",
+    "    Returns: logbook dataframe with valid datatime\n",
+    "    \"\"\"\n",
+    "    for line in range(1,len(df['Event'])+1):\n",
+    "        try:\n",
+    "            df['Start_Event'][line] = datetime.strptime(df['Start_Event'][line], '%d/%m/%Y %H:%M')\n",
+    "            df['End_Event'][line] = datetime.strptime(df['End_Event'][line], '%d/%m/%Y %H:%M')\n",
+    "        except:\n",
+    "            df = df.drop(index=line)\n",
+    "    df = df.reset_index()\n",
+    "    return df"
+   ]
+  },
```

Comment:
We can integrate this function with the `def read_logbook(path)` by adding:

```
for line in range(len(df['Event'])):
        try:
            df.Start_Event[line] = pd.to_datetime(df.Start_Event[line])
            df.End_Event[line] = pd.to_datetime(df.End_Event[line])
        except:
            df = df.drop(index=line)
    df = df.reset_index()
```
considering the date format in the .csv file yyyy/mm-dd

Reply:
Timestamp data format = '%d/%m/%Y %H:%M'

Reply:
Maybe we should use the same date format as the timestamp of the files? It is '%Y-%m-%d %H:%M'. Or we can use '%Y/%m/%d %H:%M'. I think at least day and month should be at the same position to avoid any probelem.